rust/noosphere-storage/src/implementation/indexed_db.rs

```diff
@@ -132,50 +147,105 @@ impl WebStore {
 }
 
 #[async_trait(?Send)]
-impl Store for WebStore {
+impl Store for IndexedDbStore {
     async fn read(&self, key: &[u8]) -> Result<Option<Vec<u8>>> {
         let (store, tx) = self.start_transaction(TransactionMode::ReadOnly)?;
-        let key = WebStore::bytes_to_typed_array(key)?;
+        let key = IndexedDbStore::bytes_to_typed_array(key)?;
 
-        let maybe_dag = WebStore::read(&key, &store).await?;
+        let maybe_dag = IndexedDbStore::read(&key, &store).await?;
 
-        WebStore::finish_transaction(tx).await?;
+        IndexedDbStore::finish_transaction(tx).await?;
 
         Ok(maybe_dag)
     }
 
     async fn write(&mut self, key: &[u8], bytes: &[u8]) -> Result<Option<Vec<u8>>> {
         let (store, tx) = self.start_transaction(TransactionMode::ReadWrite)?;
 
-        let key = WebStore::bytes_to_typed_array(key)?;
-        let value = WebStore::bytes_to_typed_array(bytes)?;
+        let key = IndexedDbStore::bytes_to_typed_array(key)?;
+        let value = IndexedDbStore::bytes_to_typed_array(bytes)?;
 
-        let old_bytes = WebStore::read(&key, &store).await?;
+        let old_bytes = IndexedDbStore::read(&key, &store).await?;
 
         store
             .put(&value, Some(&key))
             .await
             .map_err(|error| anyhow!("{:?}", error))?;
 
-        WebStore::finish_transaction(tx).await?;
+        IndexedDbStore::finish_transaction(tx).await?;
 
         Ok(old_bytes)
     }
 
     async fn remove(&mut self, key: &[u8]) -> Result<Option<Vec<u8>>> {
         let (store, tx) = self.start_transaction(TransactionMode::ReadWrite)?;
 
-        let key = WebStore::bytes_to_typed_array(key)?;
+        let key = IndexedDbStore::bytes_to_typed_array(key)?;
 
-        let old_value = WebStore::read(&key, &store).await?;
+        let old_value = IndexedDbStore::read(&key, &store).await?;
 
         store
             .delete(&key)
             .await
             .map_err(|error| anyhow!("{:?}", error))?;
 
-        WebStore::finish_transaction(tx).await?;
+        IndexedDbStore::finish_transaction(tx).await?;
 
         Ok(old_value)
     }
 }
+
+#[cfg(feature = "performance")]
+fn js_value_to_error(value: wasm_bindgen::prelude::JsValue) -> anyhow::Error {
```

Comment:
Maybe some `impl From<JsValue> for OurError` is called for here..? Then `OurError` can have some means of being `anyhow::Error`-compatible.